src/context/dancer/types.ts

```diff
@@ -12,11 +12,26 @@ export type Dancer = {
   primaryMachine: string;
 };
 
+export type AllDancers = {
+  id: string;
+  authenticationId: string;
+  ddrName: string;
+  ddrCode: string;
+  primaryMachineLocation: string;
+  profilePictureUrl: string;
+  state: string;
+};
+
 export type DancersRepository = {
+  getAll: () => Promise<Result<Error, Array<AllDancers>>>;
```

Comment:
This can be Array<Dancer>, the object should be the same structure (can also remove the AllDancers definition from above it).